### ISSUE TITLE
How to improve custom-trained model?

### ISSUE THREAD

**Issue by bit-scientist** [ultralytics/yolov5#9795]
I trained my custom dataset with the following command: `python train.py --data data/custom_data.yaml --weights '' --cfg yolov5m6.yaml  --img 1280 --batch-size 16 --device 0`, it finished with an early stopping at epoch 160. I chose to train from scratch because my dataset domain is different from existing pre-trained models' datasets.

---
YOLOv5m6 summary:
```
     Class     Images     Labels          P          R     mAP@.5 mAP@.5:.95: 
       all        432        432      0.838      0.675      0.736      0.484
  ink_drop        432         54       0.78      0.527      0.605      0.498
line_break        432        216      0.991      0.727       0.76      0.281
      hole        432         54      0.982          1      0.995      0.871
     stain        432        108      0.598      0.444      0.583      0.287
```
---
**results.png** is as follows:
---
![image](https://user-images.githubusercontent.com/26497571/195514436-be2a0bba-f28f-46c6-b0c2-0375ef32eae2.png)
---
There is a clear indication of overfitting as the gap between `train/box_loss` and `val/box_loss` got bigger and bigger. 
Then I thought of increasing the default `weight_decay: 0.0005` to `0.0015` or even bigger, but I'm not sure if it's correct way to do this. @glenn-jocher, could you have a look at this?

What I don't understand are `val/obj_loss` and `val/cls_loss`. What can I infer from these graphs?

Also, I don't understand why precision, recall and mAP metrics started to decrease after epoch ~80?

Could it be because of my dataset's nature? My dataset consists of 4 classes (defects on fabric) and they are significantly small compared to `img-size` (1280). Initially, the original image-size was 4K image and I cropped it into 9 tiles to preserve small object visibility. Also I had around 140 defect instances in total. There is mostly 1 instance per image, so there were around 110 images in total. Therefore, I decided to augment it by segmenting defect's area and pasting it to new design materials, since most of the fabric materials don't have any defects. It's very rare that defects appear on fabrics, so I wanted to have many background images as @glenn-jocher recommended [here](https://github.com/ultralytics/yolov5/issues/9456#issuecomment-1250149064), #9413. This way, I could increase the dataset up to 45,000 for training (**is it too much?**) , and ~500 for test and validation splits each. But, although the amount of dataset is 45k now, the exact number of defects (140) didn't change at all. They are essentially all on the same location of an image, but different designs. 

What should I do to increase the dataset in a better way, hence improve the overall accuracy of the model?

And I should note that, I did review the [Tips-for-Best-Training-Results](https://github.com/ultralytics/yolov5/wiki/Tips-for-Best-Training-Results) and did my best to follow them. 

I hope to get some advice from anyone who faced similar issues.
Thank you.

**Comment by glenn-jocher** [comment]
@bit-scientist you should always start from a pretrained model regardless of domain. Low level features like edges and colors apply across any domain. I'd start there, and if you overfit then target less epochs for training with higher augmentation.

**Comment by bit-scientist** [comment]
thank you a lot, @glenn-jocher. How about increasing `weight_decay` as stated above? Dou think the idea of just copy-pasting the defect segmentations onto new design patterns works?

**Comment by glenn-jocher** [comment]
@bit-scientist it's up to you to experiment, I can't provide any insights in your specific domain.
